src/SelectPopover/SelectPopover.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { SelectPopoverItem, StyledSelectPopover, StyledSelectPopoverWrapper } from './selectPopoverStyles';
+import { createPortal } from 'react-dom';
+
+export interface SelectPopoverOption {
+    name: string;
+    value: unknown;
+    key?: string;
+}
+
+export interface SelectPopoverProps {
+    /**
+     * Called on click with the value of the option that was clicked.
+     */
+    handleClick(value: unknown): void;
+    /**
+     * Options to be mapped to items in the dropdown.
+     */
+    options: SelectPopoverOption[];
+    /**
+     * Determines whether the popover is visible. A key can be provided if the name is not unique.
+     */
+    visible: boolean;
+    /**
+     * When true, the component will mount a div to the body and render the popover through it.
+     * This is useful when the popover would be inside a scrollable container or one with "overflow: hidden"
+     * so it doesn't get cut off.
+     */
+    renderInPortal?: boolean;
+}
+
+const SelectPopover: FC<SelectPopoverProps> = (
+    {
+        children,
+        options,
+        handleClick,
+        renderInPortal = false,
+        visible = false
+    }) => {
+    const [portal, setPortal] = useState<HTMLElement | null>(null);
+    const [popoverElement, setPopoverElement] = useState<HTMLElement | null>(null);
+
+    useEffect(() => {
+        const bodyElementHTMLCollection = document.getElementsByTagName('body');
+        const bodyElement = bodyElementHTMLCollection.length > 0 ? bodyElementHTMLCollection.item(0) : null;
+
+        if (renderInPortal && bodyElement) {
+            const portalElement = document.createElement('div');
+
+            if (!portal) {
+                bodyElement.appendChild(portalElement);
+            }
+
+            if (popoverElement) {
+                if (portal) {
```

Comment:
Could this be simplified to a single if with if (popoverElement && portal)?

Reply:
Done.